android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java

```diff
@@ -259,16 +253,10 @@ private String getReasonPhraseFromResponseCode(int code) {
     }
 
     private WebResourceResponse handleCapacitorHttpRequest(WebResourceRequest request) throws IOException {
-        boolean isHttps =
-            request.getUrl().getPath() != null && request.getUrl().getPath().startsWith(Bridge.CAPACITOR_HTTPS_INTERCEPTOR_START);
-
-        String urlString = request
-            .getUrl()
-            .toString()
-            .replace(bridge.getLocalUrl(), isHttps ? "https:/" : "http:/")
-            .replace(Bridge.CAPACITOR_HTTP_INTERCEPTOR_START, "")
-            .replace(Bridge.CAPACITOR_HTTPS_INTERCEPTOR_START, "");
-        urlString = URLDecoder.decode(urlString, "UTF-8");
+        Uri uri = Uri.parse(request.getUrl().toString());
+        // `getQueryParameter` method of Android Uri lib already decodes the value automatically
+        String urlString = uri.getQueryParameter(Bridge.CAPACITOR_HTTP_INTERCEPTOR_URL_PARAM);
+
```

Comment:
```suggestion
        String urlString = request.getUrl().getQueryParameter(Bridge.CAPACITOR_HTTP_INTERCEPTOR_URL_PARAM);
```
`request.getUrl()` already returns an `Uri`, there is no need to turn it into string to create a new `Uri`